portfolio/src/main/java/com/google/sps/controller/RequestCounterServlet.java

```diff
@@ -0,0 +1,46 @@
+package com.google.sps.controller;
+
+import javax.servlet.ServletConfig;
+import javax.servlet.ServletContext;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.io.PrintWriter;
+
+@WebServlet("/requestCounterServlet")
+public class RequestCounterServlet extends HttpServlet {
+    public void init(ServletConfig config) throws ServletException {
+        super.init(config);
+
+        int count = 0;
+        // save counter to the application scope
+        getServletContext().setAttribute("counter", count);
+    }
+
+    protected void doGet(HttpServletRequest request,
+                         HttpServletResponse response) throws
+                                                                                   ServletException,
+                                                                                   IOException {
+        ServletContext servletContext = getServletContext();
+        // get the counter
```

Comment:
Comments like this are typically unnecessary since the line of code is actually quite readable! I can already tell what the code does without the comment.

Comments describing behavior like this are also error prone because someone might come along and change the code and not the comment or vice versa.

Same goes for most of the comments after this one, like `// increment the counter`

TL;DR - Readable code is usually self-documenting.

Reply:
Oh, I see, that makes a lot of sense! Thank you for helping me improve🙏🙏🙏